core/common/src/main/java/alluxio/security/authentication/AuthenticationUtils.java

```diff
@@ -0,0 +1,138 @@
+      PropertyKey.NETWORK_CONNECTION_AUTH_TIMEOUT);
+
+  /**
+   * @param subject the subject to use (can be null)
+   * @param conf Alluxio configuration
+   * @return the configured impersonation user, or null if impersonation is not used
+   */
+  @Nullable
+  public static String getImpersonationUser(Subject subject, AlluxioConfiguration conf) {
+    // The user of the hdfs client
+    String hdfsUser = null;
+
+    if (subject != null) {
+      // The HDFS client uses the subject to pass in the user
+      Set<CurrentUser> user = subject.getPrincipals(CurrentUser.class);
+      LOG.debug("Impersonation: subject: {}", subject);
+      if (!user.isEmpty()) {
+        hdfsUser = user.iterator().next().getName();
+      }
+    }
+
+    // Determine the impersonation user
+    String impersonationUser = null;
+    if (conf.isSet(PropertyKey.SECURITY_LOGIN_IMPERSONATION_USERNAME)) {
+      impersonationUser = conf.getString(PropertyKey.SECURITY_LOGIN_IMPERSONATION_USERNAME);
+      LOG.debug("Impersonation: configured: {}", impersonationUser);
+      if (Constants.IMPERSONATION_HDFS_USER.equals(impersonationUser)) {
+        // Impersonate as the hdfs client user
+        impersonationUser = hdfsUser;
+      } else {
+        // do not use impersonation, for any value that is not _HDFS_USER_
+        if (impersonationUser != null && !impersonationUser.isEmpty()
+            && !Constants.IMPERSONATION_NONE.equals(impersonationUser)) {
+          LOG.warn("Impersonation ignored. Invalid configuration: {}", impersonationUser);
+        }
+        impersonationUser = null;
+      }
+      if (impersonationUser != null && impersonationUser.isEmpty()) {
+        impersonationUser = null;
+      }
+    }
+    LOG.debug("Impersonation: hdfsUser: {} impersonationUser: {}", hdfsUser, impersonationUser);
+    return impersonationUser;
+  }
+
+  /**
+   * It builds an authenticated channel.
+   * @param connection Grpc connection
+   * @param subject subject
+   * @param authType authentication type
+   * @return AuthenticatedChannelClientDriver
+   */
+  public static AuthenticatedChannelClientDriver authenticate(
+      GrpcConnection connection, Subject subject, AuthType authType) throws AlluxioStatusException {
+    if (authType != AuthType.SIMPLE && authType != AuthType.CUSTOM) {
+      throw new UnauthenticatedException(
+          String.format("Channel authentication scheme not supported: %s", authType.name()));
+    }
+
+    GrpcChannelKey channelKey = connection.getChannelKey();
+    SaslClientHandler clientHandler =
+        new alluxio.security.authentication.plain.SaslClientHandlerPlain(
+            subject, Configuration.global());
+    AuthenticatedChannelClientDriver authDriver;
+    try {
+      // Create client-side driver for establishing authenticated channel with the target.
+      authDriver = new AuthenticatedChannelClientDriver(
+          clientHandler, channelKey);
+    } catch (SaslException t) {
```

Comment:
When the internal exceptions do not provide or cannot provide enough information, it sounds reasonable to catch and re-throw exceptions.
I saw that we rethrow exceptions here, but not covering the following operations including `startAuthenticatedChannel`. Will those operations' error/exception message be unclear for users or debuggers?

Reply:
`startAuthenticatedChannel` throws `AlluxioStatusException`. I'm assuming if a function is throwing a user facing exception, then that exception is already in a user-friendly format. Otherwise all functions will have to catch and rethrow exceptions to be "safe".

Reply:
so RuntimeException itself should be user-friendly and contains information clear enough for the end-user.
@ssz1997 FYI